### PR TITLE
Improve Error Message When wal_dir doesn't exist

### PR DESCRIPTION
**Pull request by siying**
Summary: Right now the error mesage when options.wal_dir doesn't exist is not helpful to users. Be more specific

Test Plan: Manually test the wal_dir doesn't exist case and observe the error message.

### CODE REVIEW

File: db/db_impl_open.cc [hunk 1]
```diff
@@ -422,7 +422,8 @@ Status DBImpl::Recover(
     std::vector<std::string> filenames;
     s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
     if (!s.ok()) {
-      return s;
+      return Status::InvalidArgument("wal_dir not found",
```
Comment (riversand963) [inline db/db_impl_open.cc]: `NotFound` or `InvalidArgument`?

Comment (riversand963) [reply db/db_impl_open.cc]: Ok I see. Actually the return value of `GetChildren` can be `NotFound` or `IOError`. It's possible that we tell the user `wal_dir` is not found but the actual error is permission?

Comment (siying) [reply db/db_impl_open.cc]: Hmm, good catch. I should fix it.

**Comment by siying** [comment]
@riversand963 updated.

**Pull request closed without merging**